app/controllers/bot_instances_controller.rb

```diff
@@ -59,6 +59,46 @@ def destroy
     end
   end
 
+  def reorder
+    @bot_instance = BotInstance.find(params[:id])
+
+    unless user_signed_in? && (
+      current_user.is_owner?(@bot_instance.bot) ||
+      current_user.is_collaborator?(@bot_instance.bot) ||
+      current_user.is_admin? ||
+      current_user.is_developer?)
```

Comment:
This can just be is_admin? - the roles are orthogonal, not hierarchical. Devs who also need admin access will be given it explicitly.